Fix Hitless Update Activate Firmware Mailbox command sequence

mlvisaya opened the pull request:
During HW testing, when MCU sends Activate Firmware mailbox command to perform MCU hitless update, clearing FW_EXEC_CTRL[2] didn't trigger MCI notif_cptra_mcu_reset_req_sts interrupt to MCU. Caliptra needs to trigger the interrupt via notif0_intr_trig_r register.

After the interrupt occurs, MCU will reset itself and clears the interrupt, this will be an indication to Caliptra to clear FW_EXEC_CTRL[2].

Removed the unnecesary check RESET_STATUS.MCU_RESET_STS. This indicates that the MCU is on reset, but the transition happens so fast that the SW will not catch the transition. It is enough for Caliptra to wait on notif0_intr_trig_r getting cleared to know that MCU already reset.

mhatrevi commented on rom/dev/Cargo.toml:
```diff
@@ -25,6 +25,7 @@ caliptra-lms-types.workspace = true
 caliptra-registers.workspace = true
 caliptra-x509 = { workspace = true, default-features = false }
 ufmt.workspace = true
+ureg.workspace = true
```
Why is this change required?

mlvisaya replied:
you're right this is not needed. I'll remove.

mlvisaya committed:
Fix Hitless Update Activate Firmware Mailbox command sequence

During HW testing, when MCU sends Activate Firmware mailbox command to
perform MCU hitless update, clearing FW_EXEC_CTRL[2] didn't trigger MCI
notif_cptra_mcu_reset_req_sts interrupt to MCU. Caliptra needs to
trigger the interrupt via notif0_intr_trig_r register.

After the interrupt occurs, MCU will reset itself and clears the
interrupt, this will be an indication to Caliptra to clear
FW_EXEC_CTRL[2].

Removed the unnecesary check RESET_STATUS.MCU_RESET_STS. This indicates
that the MCU is on reset, but the transition happens so fast that the SW
will not catch the transition. It is enough for Caliptra to wait on
notif0_intr_trig_r getting cleared to know that MCU already reset.

Verify MCU FW after copying it from staging to MCU SRAM

Generate interrupt when intr_block_rf_notif0_intr_trig_r is set

intr_block_rf_notif0_intr_trig_r is a register to trigger interrupts,
therefore setting intr_block_rf_notif0_intr_trig_r should trigger an
interrupt.

When activating fw, Caliptra will request interrupt through MCI.
Caliptra will wait for Caliptra for the MCU to clear the interrupt.
In the absence, of an MCU, the integration test will clear it.

Avoid waiting for MCU reset on failed test cases and SOC only
activatation

Avoid waiting for MCU reset on failed test cases. If the parameters
passed are invalid, then it is expected that the MCU will not reset.

mlvisaya committed:
Save memory by removing some helper functions

FPGA build is failing due to ICCM size reached.
Removing some helper functions that are only used once helped with
optimizing the runtime size.

mhatrevi committed:
Merge branch 'main-2.x' into mlvisaya/fpga-debug-firmware-update

mhatrevi committed:
Merge branch 'main-2.x' into mlvisaya/fpga-debug-firmware-update
